pandas/tests/test_resample.py

```diff
@@ -2777,6 +2694,211 @@ def test_evenly_divisible_with_no_extra_bins(self):
         assert_frame_equal(result, expected)
 
 
+def assert_series_or_frame_equal(result, expected):
+    if isinstance(result, Series):
+        return assert_series_equal(result, expected)
+    if isinstance(result, DataFrame):
+        return assert_frame_equal(result, expected)
+
+
+class Base_pytest(object):
+    """
+    base class for resampling testing providing fixtures for each index type
+    - index: default index defined by _index_factory in each subclass
+    - pandas_obj: default Series and DataFrame objects
+    """
+
+    _index_fixture_start = datetime(2005, 1, 1)
+    _index_fixture_end = datetime(2005, 1, 10)
+    _index_fixture_freq = 'D'
+
+    def create_index(self, *args, **kwargs):
+        """ return the _index_factory created using the args, kwargs """
+        factory = self._index_factory()
+        return factory(*args, **kwargs)
+
+    @pytest.fixture(scope='class')
+    def index(self):
+        return self.create_index(self._index_fixture_start,
+                                 self._index_fixture_end,
+                                 freq=self._index_fixture_freq)
+
+    @pytest.fixture(params=[Series, DataFrame], scope='class')
+    def pandas_obj(self, request, index):
```

Comment:
you can *then* make another fixture like ``series_and_frame`` if you want, though I find that more confusing, rather have you use params.